lib/matplotlib/widgets.py

```diff
@@ -1507,35 +1503,31 @@ def update(self):
         return False
 
     def _get_data(self, event):
-        """Get the xdata and ydata for event, with limits"""
+        """Get the xdata and ydata for event, with limits."""
         if event.xdata is None:
             return None, None
-        x0, x1 = self.ax.get_xbound()
-        y0, y1 = self.ax.get_ybound()
-        xdata = max(x0, event.xdata)
-        xdata = min(x1, xdata)
-        ydata = max(y0, event.ydata)
-        ydata = min(y1, ydata)
+        xdata = np.clip(event.xdata, *self.ax.get_xbound())
+        ydata = np.clip(event.ydata, *self.ax.get_ybound())
         return xdata, ydata
 
     def _clean_event(self, event):
-        """Clean up an event
+        """
+        Preprocess an event:
 
-        Use prev event if there is no xdata
-        Limit the xdata and ydata to the axes limits
-        Set the prev event
+        - Replace *event* by the previous event if *event* has no ``xdata``.
+        - Clip ``xdata`` and ``ydata`` to the axes limits.
+        - Update the previous event.
         """
         if event.xdata is None:
             event = self._prev_event
         else:
             event = copy.copy(event)
         event.xdata, event.ydata = self._get_data(event)
-
         self._prev_event = event
         return event
 
     def press(self, event):
-        """Button press handler and validator"""
+        """Button press handler and validator."""
```

Comment:
Might we want to make this an actual sentence like: "Validate and handle a button press"?

Reply:
I tend to think of these callbacks as nouns ("it's the event handler for yada yada") rather than verbs ("do this and that") -- after all, users generally use them as "objects" (passing them to mpl_connect) rather than actually calling them.